src/middleware/error-handler.ts

```diff
@@ -0,0 +1,33 @@
+import createError, { HttpError } from 'http-errors';
+import { Context, Middleware, Next } from 'koa';
+import { hydra } from 'rdf-namespaces';
+
+const handleHttpError = (error: HttpError, { response }: Context): void => {
+  response.status = error.status;
+  response.body = {
+    '@type': hydra.Status,
+    [hydra.title]: { '@value': response.message, '@language': 'en' },
+  };
+
+  if (error.message !== response.message) {
```

Comment:
seems a weak check to be performed on error strings rather than on what the tests describe, which is the exception being a `non-http error`. But the type of error is `HttpError` so not sure how it got here in the first place.

Reply:
Ah there's a `toHttpError`. Maybe leave the original thrown object rather than wrapping in `HttpError`, or mark it so that it is clear it wasn't originally if it has an option for that? Otherwise I suspect throwing a 500 is indistinguishable from throwing a string (from this middleware rendering a response).

Reply:
Would you ever throw a 500?

Reply:
If there isn't a clear mapping of the server-side error to a status code; but yes, why bothering when you can just throw the right exception instead.